src/main/java/net/coderbot/iris/gui/GuiUtil.java

```diff
@@ -0,0 +1,83 @@
+package net.coderbot.iris.gui;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.DrawableHelper;
+import net.minecraft.client.sound.PositionedSoundInstance;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.util.Identifier;
+
+/**
+ * Class serving as abstraction and
+ * centralization for common GUI
+ * rendering/other code calls.
+ *
+ * Helps allow for easier portability
+ * to Minecraft 1.17 by abstracting
+ * some code that will be changed.
+ */
+public final class GuiUtil {
+	/**
+	 * Resource ID for the texture used for
+	 * Iris's GUI widgets.
+	 */
+	public static final Identifier IRIS_WIDGETS_TEX = new Identifier("iris", "textures/gui/widgets.png");
+
+	private GuiUtil() {}
+
+	private static MinecraftClient client() {
+		return MinecraftClient.getInstance();
+	}
+
+	/**
+	 * Binds the texture to be used for
+	 * succeeding draw calls.
+	 *
+	 * @param texture The resource ID of the texture to be bound.
+	 */
+	public static void bindTexture(Identifier texture) {
```

Comment:
This should probably be included within drawButton.

Reply:
There's a specific reason why i didn't call to `bindTexture` directly in `drawButton`, and that is because `drawButton` can also draw a button with the vanilla widgets texture bound to appear like a vanilla button.

Reply:
Makes sense. I think this is fine as long as the bindWidgetsTexture change is done.

Reply:
Also I believe this method is going to change in 1.17, and as something used very frequently in GUI code it would be easier to abstract it all so it can be changed quickly when porting